ctru-rs/src/mii.rs

```diff
@@ -0,0 +1,479 @@
+        };
+
+        let mouth_details = MouthDetails {
+            style: partial_vec_to_u8_with_reverse(&raw_mouth_details[0..5]),
+            color: partial_vec_to_u8_with_reverse(&raw_mouth_details[6..8]),
+            scale: partial_vec_to_u8_with_reverse(&raw_mouth_details[9..12]),
+            y_scale: partial_vec_to_u8_with_reverse(&raw_mouth_details[13..15]),
+        };
+
+        let mustache_details = MustacheDetails {
+            mouth_y_position: partial_vec_to_u8_with_reverse(&raw_mustache_details[0..4]),
+            mustache_style: partial_vec_to_u8_with_reverse(&raw_mustache_details[5..7]),
+        };
+
+        let beard_details = BeardDetails {
+            style: partial_vec_to_u8_with_reverse(&raw_beard_details[0..2]),
+            color: partial_vec_to_u8_with_reverse(&raw_beard_details[3..5]),
+            scale: partial_vec_to_u8_with_reverse(&raw_beard_details[6..9]),
+            y_position: partial_vec_to_u8_with_reverse(&raw_beard_details[10..14]),
+        };
+
+        let glass_details = GlassDetails {
+            style: partial_vec_to_u8_with_reverse(&raw_glass_details[0..3]),
+            color: partial_vec_to_u8_with_reverse(&raw_glass_details[4..6]),
+            scale: partial_vec_to_u8_with_reverse(&raw_glass_details[7..10]),
+            y_position: partial_vec_to_u8_with_reverse(&raw_glass_details[11..15]),
+        };
+
+        let mole_details = MoleDetails {
+            is_enabled: raw_mole_details[0],
+            scale: partial_vec_to_u8_with_reverse(&raw_mole_details[1..4]),
+            x_position: partial_vec_to_u8_with_reverse(&raw_mole_details[5..9]),
+            y_position: partial_vec_to_u8_with_reverse(&raw_mole_details[10..14]),
+        };
+
+        MiiData {
+            mii_options: options,
+            mii_selector_position: position,
+            mii_console_identity: device,
+            system_id,
+            mac_address: creator_mac,
+            mii_details: details,
+            mii_name,
+            height,
+            width,
+            face_style,
+            face_details,
+            hair_style,
+            hair_details,
+            eye_details,
+            eyebrow_details,
+            nose_details,
+            mouth_details,
+            mustache_details,
+            beard_details,
+            glass_details,
+            mole_details,
+            author_name,
+        }
+    }
+}
+
+// Methods to handle "_bits_", ``bitvec`` cannot compile to 32-bit targets, so I had to create a few
+// helper methods
+
+/// Transforms a u8 into a [bool; 8]
+fn vec_bit(data: u8) -> [bool; 8] {
+    (0..8)
+        .map(|i| (data & (1 << i)) != 0)
+        .collect::<Vec<bool>>()
+        .try_into()
+        .unwrap()
+}
+
+/// Transforms a [bool; 8] into an u8
+fn vec_bit_to_u8(data: [bool; 8]) -> u8 {
+    let mut result: u8 = 0;
+    data.map(u8::from).iter().for_each(|&bit| {
+        result <<= 1;
+        result ^= bit;
+    });
+    result
+}
```

Comment:
nit: This could be simplified:
```suggestion
fn vec_bit_to_u8(data: [bool; 8]) -> u8 {
    data.into_iter().fold(0, |result, bit| {
        (result << 1) ^ u8::from(bit)
    })
}
```